### PR TITLE
Feat/infinity scroll

### PR DESCRIPTION
**Pull request by calilrenner**
- [x] On all screens that have posts loading, start searching the API for only 10 posts at a time, loading more posts as the user scrolls to the bottom of the screen. Tip: search for a library called react-infinite-scroller

### CODE REVIEW

**Review by thipereira02 — APPROVED** [review]

**Review by thipereira02 — CHANGES_REQUESTED** [review]
There are conflicts to be solved

**Comment by calilrenner** [comment]
fixed conflicts

File: src/App.js [hunk 1]
```diff
@@ -11,6 +11,7 @@ import MyLikes from "./pages/MyLikes";
 import UserPosts from "./pages/UserPosts";
 
 export default function App() {
+  console.warn = () => {};
```
Comment (sidneyouteiro) [inline src/App.js]: Remover console

File: src/pages/Hashtag.js [hunk 2]
```diff
@@ -6,35 +6,94 @@ import Post from "../components/Post";
...
+  const [newHashTagPosts, setNewHashTagPosts] = useState([]);
+  const [hasMore, setHasMore] = useState(true);
+
+  function postRepost(post) {
+    if(post.repostId) {
+      return post.repostId;
+    } else {
+      return post.id;
+    }
+  }
```
Comment (sidneyouteiro) [inline src/pages/Hashtag.js]:
```
  function postRepost(post) {
    return post.repostId ? post.repostId : post.id;
  }
```

File: src/pages/MyLikes.js [hunk 3]
```diff
@@ -5,37 +5,106 @@ import UserContext from "../contexts/UserContext";
...
+  const [hasMore, setHasMore] = useState(true);
+  const [newLikedPosts, setNewLikedPosts] = useState([]);
+
+  function postRepost(post) {
+    if(post.repostId) {
+      return post.repostId;
+    } else {
+      return post.id;
+    }
+  }
```
Comment (sidneyouteiro) [inline src/pages/MyLikes.js]: E essa função é criada/utilizada em diversos componentes.
Acredito poderíamos a colocar em algum arquivo e importa-lá conforme a necessidade
```
 function postRepost(post) {
    return post.repostId ? post.repostId : post.id;
  }
```

File: src/pages/MyPosts.js [hunk 4]
```diff
@@ -5,42 +5,113 @@ import { getUserPosts } from "../service/api.service";
...
+  const [newUserPosts, setNewUserPosts] = useState([]);
+  const [hasMore, setHasMore] = useState(true);
+
+  function postRepost(post) {
+    if(post.repostId) {
+      return post.repostId;
+    } else {
+      return post.id;
+    }
+  }
```
Comment (sidneyouteiro) [inline src/pages/MyPosts.js]:
```
  function postRepost(post) {
    return post.repostId ? post.repostId : post.id;
  }
```

File: src/pages/Timeline.js [hunk 5]
```diff
@@ -46,37 +65,85 @@ export default function Timeline() {
...
+      loadMorePosts(firstPostId, userData.token).then(r => {
+        setNewPosts([...r.data.posts])
+        setHasMore(newPosts.length > 0)
+        setFollowedPosts([...followedPosts, ...newPosts]);
+      })
     }
 
-  function loadPosts() {
-    if (followedPosts === "") {
+    console.log(followedUsers)
```
Comment (sidneyouteiro) [inline src/pages/Timeline.js]: Remover console.log

File: src/pages/UserPosts.js [hunk 6]
```diff
@@ -21,22 +23,78 @@ export default function UserPosts() {
...
-    );
-    getFollows(userData.token).then((r) => setFollows(r.data.users));
-  }, [following, id]);
+  function postRepost(post) {
+    if(post.repostId) {
+      return post.repostId;
+    } else {
+      return post.id;
+    }
+  }
```
Comment (sidneyouteiro) [inline src/pages/UserPosts.js]:
```
  function postRepost(post) {
    return post.repostId ? post.repostId : post.id;
  }
```

**Review by sidneyouteiro — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 1 above)
Comment (calilrenner) [reply src/App.js]: Console removed

File: src/pages/Hashtag.js (hunk 2 above)
Comment (calilrenner) [reply src/pages/Hashtag.js]: I've put the this sentence directly inside useEffect callback.

File: src/pages/MyLikes.js (hunk 3 above)
Comment (calilrenner) [reply src/pages/MyLikes.js]: I've put the this sentence directly inside useEffect callback.

File: src/pages/MyPosts.js (hunk 4 above)
Comment (calilrenner) [reply src/pages/MyPosts.js]: I've put the this sentence directly inside useEffect callback.

File: src/pages/UserPosts.js (hunk 6 above)
Comment (calilrenner) [reply src/pages/UserPosts.js]: I've put the this sentence directly inside useEffect callback.

**Review by sidneyouteiro — APPROVED** [review]

**Pull request merged**